examples/to_mdbx.rs

```diff
@@ -0,0 +1,150 @@
+//! Convert a block folder to a MDBX folder
+//! For help, run: `cargo run --example to_mdbx -- --help`
+
+use alloy_primitives::{Address, Bytes, FixedBytes, B256, B64, U64};
+use clap::Parser;
+use libmdbx::{
+    Database, DatabaseKind, DatabaseOptions, Mode, NoWriteMap, ReadWriteOptions, SyncMode,
+    TableFlags, WriteFlags,
+};
+use pevm::{EvmAccount, EvmCode};
+use revm::primitives::Bytecode;
+use std::{
+    collections::{BTreeMap, HashMap},
+    fs::File,
+    io::{BufReader, Error},
+    path::Path,
+};
+
+type B416 = FixedBytes<52>;
+type B0 = FixedBytes<0>;
+
+/// Convert a block folder to a MDBX folder
+#[derive(Parser, Debug)]
+#[clap(name = "to_mdbx")]
+struct Args {
+    /// Path to bytecodes.bincode
+    #[clap(long, value_name = "FILE")]
+    bytecodes: String,
+    /// Path to pre_state.json
+    #[clap(long, value_name = "FILE")]
+    pre_state: String,
+    /// Path to block_hashes.json
+    #[clap(long, value_name = "FILE")]
+    block_hashes: Option<String>,
+    /// Path to output MDBX dir
+    #[clap(long, value_name = "DIR")]
+    output: String,
+}
+
+#[derive(Debug, Clone, Default)]
+struct Tables {
+    balance: HashMap<Address, B256>,
+    nonce: HashMap<Address, B64>,
+    code_hash: HashMap<Address, B256>,
+    code_by_hash: HashMap<B256, Bytes>,
+    has_storage: HashMap<Address, B0>,
+    storage: HashMap<B416, B256>,
+    block_hash: HashMap<B64, B256>,
+}
+
+fn open_db(dir: impl AsRef<Path>) -> Result<Database<NoWriteMap>, libmdbx::Error> {
+    const MB: isize = 1048576;
+
+    #[allow(clippy::identity_op)]
+    Database::<NoWriteMap>::open_with_options(
```

Comment:
Will `open` work for now? Ideally we'll have a dedicated task/PR for option tuning.

Reply:
`open` does not work, it will cause error `MDBX_MAP_FULL`.

By default, the max num tables is 1 (the default table), and the max db file size is a few of MBs (which is too small).

fyi, https://github.com/erthink/libmdbx/issues/136

Reply:
I see, then let's put the options in a `const`/`static` with a comment on each custom config on why it's there.